services/UserService.js

```diff
@@ -0,0 +1,33 @@
+import { UserDao } from '../models';
+import bcrypt from 'bcrypt';
+import jwt from 'jsonwebtoken';
+
+const { JWT_SECRET_KEY } = process.env;
+
+const logIn = async (userAccount, password) => {
+  const userInfo = await UserDao.findUser(userAccount);
+  let err;
+
+  if (!userInfo.length) {
+    err = new Error('INVALID_USER');
+    err.statusCode = 400;
+    throw err;
+  }
+
+  const { user_account, password: hashedPassword } = userInfo[0];
+  const isMatch = await bcrypt.compare(password, hashedPassword);
+
+  if (!isMatch) {
+    err = new Error('INVALID_USER');
+    err.statusCode = 400;
+    throw err;
+  }
+
+  const access_token = jwt.sign({ user_account }, JWT_SECRET_KEY, {
+    expiresIn: '30m',
+  });
+
```

Comment:
```suggestion

  const access_token = jwt.sign({ id }, JWT_SECRET_KEY, {
    expiresIn: '30m',
  });

```
토큰에 email을 담는 것은 조금 위험합니다.
토큰은 (정확히는 token의 payload 부분은) 외부에서 쉽게 열어서 정보를 확인하는 것이 가능하기 때문에 사용자를 너무 명확히 식별할 수 있는 이메일을 넣으면 개인 정보를 안전히 지키기가 어렵습니다. 사실 요즘엔 융통성있게 이메일을 사용하는 회사도 많아졌습니다 :) 학습의 차원에서 조금 더 안전하게 사용자의 정보를 정확하게 알 수 없는 database id를 넣어주시는 것이 더 좋겠습니다 👍

Reply:
이메일을 담으면 위험하군요! 감사합니다!
database id를 넣어주시는 것이 더 좋다고 하셨는데 database id라는 것이
user 테이블의 PK인 id를 말씀하시는 것인지
아니면 유저가 로그인 시 입력하는 그 id를 말씀하시는 것인지
아니면 사실은 database에도 id가 존재하는데 그 database의 id를 말씀하시는 것인가요?

그리고 제가 저기서 담아준 user_account는 email이 아닌 유저가 로그인할 때 사용하는 ID입니다!
email은 별도로 존재합니다!

![image](https://user-images.githubusercontent.com/50080535/125375639-6145ed80-e3c4-11eb-81d8-4ce35e322921.png)

Reply:
소헌님 감사합니다! 줌에서 답변해주신 내용을 바탕으로
user 테이블의 id(PK)를 payload에 담아주었습니다!

https://github.com/wecode-bootcamp-korea/fullstack1-1st-BaseNote-backend/pull/2/commits/d51d474a8f258b888ec8f20f196a24ddb6d28efe